docs/security/trust/trust_delegation.md

```diff
@@ -0,0 +1,222 @@
+available on your path.
+
+For more detailed information about how to use Notary outside of the default
+Docker Content Trust use cases, please read the
+[the Notary CLI documentation](https://github.com/docker/notary/blob/master/docs/getting_started.md).
+
+Note that when publishing and listing delegation changes using the Notary client,
+your Docker Hub credentials are required.
+
+## Generating delegation keys
+
+Your collaborator needs to generate a private key (either RSA or ECDSA)
+and give you the public key so that you can add it to the `targets/releases`
+delegation.
+
+The easiest way to for them to generate these keys is with OpenSSL.
+Here is an example of how to generate a 2048-bit RSA portion key (all RSA keys
+must be at least 2048 bits):
+
+```
+$ opensl genrsa -out delegation.key 2048
+Generating RSA private key, 2048 bit long modulus
+....................................................+++
+............+++
+e is 65537 (0x10001)
+
+```
+
+They should keep `delegation.key` private - this is what they will use to sign
+tags.
+
+Then they need to generate a x509 certificate containing the public key, which is
+what they will give to you.  Here is the command to generate a CSR (certificate
+signing request):
+
+```
+$ openssl req -new -sha256 -key delegation.key -out delegation.csr
+```
+
+Then they can send it to whichever CA you trust to sign certificates, or they
+can self-sign the certificate (in this example, creating a certificate that is
+valid for 1 year):
+
+```
+$ openssl x509 -req -days 365 -in delegation.csr -signkey delegation.key -out delegation.crt
+```
+
+Then they need to give you `delegation.crt`, whether it is self-signed or signed
+by a CA.
+
+## Adding a delegation key to an existing repository
+
+Prior to Docker Engine 1.11, the snapshot key was generated and managed locally.
+Before adding any delegations, you should rotate the snapshot key to the server
+so that collaborators will not require your snapshot key to sign and publish tags:
+
+```
+$ notary -d ~/.docker/trust key rotate docker.io/<username>/<imagename> snapshot -r
+```
+
+This tells Notary to rotate a key for your particular image repository - note that
+you must include the `docker.io/` prefix.  `-d ~/.docker/trust` tells Notary where your
+local Content Trust data is, and `snapshot -r` specifies that you want to rotate
+the snapshot key specifically, and you want the server to manage it (`-r`
+stands for "remote").
+
+When adding a delegation, your must acquire
+[the PEM-encoded x509 certificate with the public key](#generating-delegation-keys)
+of the collaborator you wish to delegate to.
+
+Assuming you have the certificate `delegation.crt`, you can add a delegation
+for this user and then publish the delegation change:
+
+```
+$ notary -d ~/.docker/trust delegation add docker.io/<username>/<imagename> targets/releases delegation.crt --all-paths
```

Comment:
Ah ok, notary has a separate default configuration default directory, since it can be used on its own without docker, so it has to be configured to point to docker.  I'll have the users set a `~/notary/config.json` so we don't have to include flag.